### PR TITLE
Reverse 1:n relationships

### PR DESCRIPTION
**Pull request by lutter**
This pull request reverse how 1:n associations are stored in Uniswap: rather than storing them as an array on the 'n' side, they are now stored as a scalar attribute on the '1' side. Doing it this way makes things much easier for the database when executing queries.

One downside is that changing which side is stored and which side is derived changes what filters are available - if that conflicts with how Uniswap needs to be queried, feel free to ignore any of this.

I did sync with this version of Uniswap on a test system successfully, but I don't know the ins and outs of the subgraph enough to be able to tell whether I broke some other functionality.

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -60,7 +60,7 @@ type Exchange @entity {
     weightedAvgPriceUSD: BigDecimal!        # weightAvgPriceUSD = ( $1 of ETH in ETH ) / weightedAvgPrice
 
     # Fields used to help derived relationship
-    factory: Uniswap! @derivedFrom(field: "exchanges")
+    factory: Uniswap
```
Comment (davekaj) [inline schema.graphql:63]: We can add a `!` to this

**Comment by davekaj** [comment]
Merging this would close https://github.com/graphprotocol/uniswap-subgraph/issues/28

**Comment by ianlapham** [comment]
This makes sense to me - if the performance would greatly improve it might be worth it to update our queries on our frotn-end to support the changes. Is there a way to deploy this to some other subgraph so I could test it first?

**Comment by davekaj** [comment]
I just deployed it, but I assume lutter already deployed it somewhere because it was instantly synced for me. You can see it here https://thegraph.com/explorer/subgraph/davekaj/uniswap?version=pending

**Comment by lutter** [comment]
Yes, I had deployed it a while ago, and completely forgotten about it; it's also at https://thegraph.com/explorer/subgraph/lutter/uniswap-rels

**Comment by ianlapham** [comment]
I just tested this with our v1 info client - it actually seems to work with no changes required to our queries. If this helps us sync faster through grafting I'm ok to merge this in. @davekaj @lutter

**Comment by davekaj** [comment]
great, will merge now

**Pull request merged**